sdk/storage/azure-storage-file-share/src/test/java/com/azure/storage/file/share/FileServiceAsyncApiTests.java

```diff
@@ -143,12 +144,14 @@ public void listSharesWithFilter(ListSharesOptions options, int limits, boolean
                 share.setMetadata(TEST_METADATA);
             }
             testShares.add(share);
-            primaryFileServiceAsyncClient
-                .createShareWithResponse(share.getName(), share.getMetadata(), share.getProperties().getQuota())
-                .block();
         }
 
-        StepVerifier.create(primaryFileServiceAsyncClient.listShares(options.setPrefix(prefix)))
+        Mono<Void> createShares = Flux.fromIterable(testShares)
+            .flatMap(share -> primaryFileServiceAsyncClient.createShareWithResponse(share.getName(),
+                share.getMetadata(), share.getProperties().getQuota()))
+            .then();
```

Comment:
`then()` is also useful for saying "when all operations in the `Flux` complete", but this could be removed as `thenMany` does the same thing: https://projectreactor.io/docs/core/release/api/reactor/core/publisher/Flux.html#thenMany-org.reactivestreams.Publisher-